src/frontend/src/helpers/Storage/LocalStorage.ts

```diff
@@ -269,96 +299,136 @@ class LocalStorage implements AbstractStorage {
+      return log || false;
+    });
   }
 
   public async pushChangeLog(change: ChangeLog): Promise<number> {
+    this.debug && console.log(`pushChangeLog`);
     const top = await this.topChangeLog();
-    if (top && change.type === "editFile" && top.type === "editFile") {
-      await this.popChangeLog();
-      top.contents = change.contents;
-    }
-    return await this.db.changeLogs.put(change);
+    const tbs = [this.db.files, this.db.projects, this.db.settings, this.db.changeLogs];
+    return await this.db.transaction("rw", tbs, async () => {
+      if (top && change.type === "editFile" && top.type === "editFile") {
+        await this.popChangeLog();
+        top.contents = change.contents;
+      }
+      return await this.db.changeLogs.put(change);
+    });
   }
 
   public async popChangeLog(): Promise<ChangeLog|false> {
-    const top = await this.topChangeLog();
-    if (top) {
-      if (top.id) {
-        await this.db.changeLogs.delete(top.id);
-        return top;
+    this.debug && console.log(`popChangeLog`);
+    const tbs = [this.db.files, this.db.projects, this.db.settings, this.db.changeLogs];
+    return await this.db.transaction("rw", tbs, async () => {
+      const top = await this.topChangeLog();
+      if (top) {
+        if (top.id) {
+          await this.db.changeLogs.delete(top.id);
+          return top;
+        } else {
+          return false;
+        }
       } else {
         return false;
       }
-    } else {
-      return false;
-    }
+    });
   }
 
   public async countChangeLogs(): Promise<number> {
-    return await this.db.changeLogs.count();
+    this.debug && console.log(`countChangeLogs`);
+    const tbs = [this.db.files, this.db.projects, this.db.settings, this.db.changeLogs];
+    return await this.db.transaction("rw", tbs, async () => {
+      return await this.db.changeLogs.count();
+    });
   }
 
   public async clearChangeLogs(): Promise<void> {
-    this.db.changeLogs.clear();
+    this.debug && console.log(`clearChangeLogs`);
+    const tbs = [this.db.files, this.db.projects, this.db.settings, this.db.changeLogs];
+    return await this.db.transaction("rw", tbs, async () => {
+      this.db.changeLogs.clear();
+    });
   }
 
   // Will be replaced by Dexie.Syncable.ISyncProtocol
   public async applyChanges(changeLogs: ChangeLog[],
                             newProjects: string[],
                             deletedProjects: ProjectID[]): Promise<void> {
-    const tbs = [this.db.files, this.db.changeLogs, this.db.projects, this.db.settings];
-    return await this.db.transaction("rw", tbs, async () => {
-      Dexie.currentTransaction.on("abort", () => {
-        console.warn("applyChanges transaction aborted");
-      });
+    this.debug && console.log(`applyChanges`);
+    const tbs = [this.db.files, this.db.projects, this.db.settings, this.db.changeLogs];
+    // return await this.db.transaction("rw", tbs, async () => {
```

Comment:
This needs to be in a _single_ transaction, as otherwise you will run into an inconsistent database.